docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+
+Going forward, users of EdgeX will want to have more insights – that is more metrics and control plane events – on what is happening in the data collection, device management and monitoring, and event handling processes inside of Edge services.  
+
+### Definitions
+
+**Metric (or telemetry) data** is defined as the count or rate of some action, resource, or circumstance in the EdgeX instance or specific service.  Examples or metrics include:
+
+- the number of EdgeX Events sent from core data to an application service via ZeroMQ
+- the number of requests on a service API
+- the average time it takes to process a message through an application service
+- The number of errors logged by a service
+
+**Control plane events** (CPE) are defined as discrete system or service incidents (or `events`) that occur within an EdgeX instance.  Examples of CPE include:
+
+- a device was provisioned (added to core metadata)
+- a service was stopped
+- service configuration has changed
+
+Note CPE should not be confused with core data Events.  Core data Events represent a collection (one or more) of sensor/device readings.  Core data Events represent sensing of some measured state of the physical world (temperature, vibration, etc.).  CPE represents the detection of some happening inside of the EdgeX software.
+
+The collection and dissemination of metric data and CPE will require service level instrumentation relevant to capture and send data about relevant EdgeX operations.  EdgeX does not currently offer any service instrumentation.
+
+### Metric and CPE Consumption/Use
+
+At this time, EdgeX will make metric data and CPE available to other subscribing 3rd party applications and systems, but will not necessarily consume or use this information itself.  
+
+In the future, EdgeX may consume its own metric data or CPE.  For example, EdgeX may, in the future,
+
+- use a metric on the number of EdgeX events being sent to core data as the means to throttle back device data collection
+- use a configuration change CPE to automatically restart a service.
+
+In the future, EdgeX application services may optionally subscribe to metrics and CPE messages (by attaching to the appropriate message pipe).  Thus allowing additional filtering, transformation, endpoint control of metric and CPE data.
+
+- At the point where this feature is supported, consideration would need to be made as to whether all events (sensor reading messages and metric/CPE messages go through the same application services)

```

Comment:
wrong parenthesis position.
it should be
whether all events (sensor reading messages and metric/CPE messages**)** go through the same application services